# MAP paper project- 
## checking the 2012 Nature paper LYS and CYS positions to see how many overlap with positions from UKB_CCDSset file and how many are not in the CCDS 2018 file
- either the positions are not there because the ID has been updated or dropped

original attempt = 8.12.19

8.26.19

mfpfox


### directory of project: "/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/"

## input files: 
1. LYS_reactivity_2012_4579.csv
2. LYS_pan_target_2012_8360.csv
3. CYS_reactivity_2012_1541.csv
4. CYS_pan_target_2012_6157.csv
### positions of C and K in 2018 canonical sequence file from UKB

**Kpos_CCDSset.csv** == 614,333 lysines in 2018 CCDS UKB

**Cpos_CCDSset.csv** == 242,940 cysteines in 2018 CCDS UKB



# TOTAL counts: 

### total cys data before 2018 check = 7698
### total cys data after 2018 check = 7483
### total lost cys = 215

#### reactivity = 1541
#### react in 2018 pos = 1492
#### total lost react = 49
---
#### pan/target = 6157
#### pan/target in 2018 pos = 5991
#### total lost pan/tar = 166
---
---
### total lys data before 2018 check = 12939
### total lys data after 2018 check = 12724
### total lost lys = 215

#### reactivity = 4579
#### react in 2018 pos = 4504
#### total lost react = 75

#### pan/target = 8360
#### pan/target in 2018 pos = 8220
#### total lost pan/tar = 140


## total reactivity CYS LYS = 5996
## total pan/target CYS LYS = 14211




# total ever labeled cys (pan/target/reactivity all unique): 

# total ever labeled lys (pan/target/reactivity all unique): 


# REASON why some Nature positions dont match CCDS CYS LYS positions: 
1. entry ID has been updated
2. entry has been merged with another 
3. sequence corresponding to entry has changed so CYS no longer in same pos

In [2]:
import os
import sys
import pandas as pd
import numpy as np

In [3]:
os.chdir("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/")
print(os.listdir())

['Cys_ever_labeled_set_6315.csv', 'CYS_pan_target_2012_6157.csv', '.DS_Store', 'LYS_reactivity_2012_4579.csv', 'CYS_reactivity_2012_1541.csv', 'Cpos_CCDSset.csv', 'Lys_reactive_LMHset_4504.csv', 'Kpos_CCDSset.csv', 'GENECOUNT_dictionary', 'NATURE2012_UKBccds2018_files', 'Cys_panTar_5991.csv', 'LYS_pan_target_2012_8360.csv', 'Lys_panTar_8220.csv', 'Lys_panTarget_notFoundin2018positions_140pos.csv', 'Reprocesssing2012_Nature_allCYSLYSdata.xlsx', 'Cys_panTarget_notFoundin2018positions_166pos.csv', 'Cys_reactive_LMHset_1492.csv', 'Lys_ever_labeled_set_9162.csv', 'Lys_reactivity_notFoundin2018positions_75pos.csv', 'CoordinatesP06733_K64.txt', 'NOTES.sh', 'Cys_reactivity_notFoundin2018positions_49pos.csv']


# [TODO 1 ] make sure there is no redundancy in each reactivity file and each pan/target file, then do overlap with 2018 positions and keep only those that are in 2018 file

- total reactive cys in 2018 canonical seq: 
- total lost = 

- total pan/target cys in 2018 canonical seq: 
- total lost = 

- total reactive lys in 2018 canonical seq:
- total lost = 

- total pan/target lys in 2018 canonical seq: 
- total lost = 



In [4]:
# importing 2018 files
Cysnature = pd.read_csv("Cpos_CCDSset.csv", header=None)
Lysnature = pd.read_csv("Kpos_CCDSset.csv", header=None)

# importing cys react and pan/target
Creact = pd.read_csv("CYS_reactivity_2012_1541.csv",header=None)
Cpan = pd.read_csv("CYS_pan_target_2012_6157.csv",header=None)

# importing lys react and pan/target
Kreact = pd.read_csv("LYS_reactivity_2012_4579.csv",header=None)
Kpan = pd.read_csv("LYS_pan_target_2012_8360.csv",header=None)


In [5]:
Cysnature.shape
# create set 
Cysnature.columns = ['pos_ID']
Lysnature.columns = ['pos_ID']

In [6]:
# creating unique set 
Cys2018 = set(Cysnature.pos_ID)
Lys2018 = set(Lysnature.pos_ID)
print(len(Cys2018))
print(len(Lys2018))
# all the ukb positions of CYS and LYS unique as expected

242940
614333


In [7]:
# changing labels
labels = ["pos_ID","reactivity"]
Creact.columns = labels
Kreact.columns = labels
labelpan = ['pos_ID','labelType']
Kpan.columns=labelpan
Cpan.columns=labelpan

In [8]:
# drop duplicates in the reactivity files of cys and lysine
#cysreact_dup = pd.concat(d for _, d in Creact.groupby("pos_ID") if len(d) > 1)

#cysreact_dup
print(len(Creact.pos_ID))
print(len(set(Creact.pos_ID)))

print(len(Kreact.pos_ID))
print(len(set(Kreact.pos_ID)))

print(len(Cpan.pos_ID))
print(len(set(Cpan.pos_ID)))

print(len(Kpan.pos_ID))
print(len(set(Kpan.pos_ID)))

1541
1541
4579
4579
6157
6157
8360
8360


In [9]:
1541 + 6157

7698

In [10]:
4504+8220

12724

In [11]:
140+75

215

# Nature data does not have redundancy when you process the reactivity data seperate from the pan/target data

## before checking if 2012 position matches 2018: 
- total cys data before 2018 check = 7698
    - reactivity = 1541
    - react in 2018 pos = 1492
    - total lost react = 49
    
    - pan/target = 6157
    - pan/target in 2018 pos = 5991
    - total lost pan/tar = 166
- total cys data after 2018 check = 7483
- total lost cys = 215
    
- total lys data before 2018 check = 12939
    - reactivity = 4579
    - react in 2018 pos = 4504
    - total lost react = 75
    
    - pan/target = 8360
    - pan/target in 2018 pos = 8220
    - total lost pan/tar = 140
- total lys data after 2018 check = 12724
- total lost lys = 215

- total reactivity CYS LYS = 5996

In [12]:
1492 + 4504

5996

In [13]:
# finding overlap between cys and lysine reactivity with 2018 positions

# making list of react pos_ID column
CreactList = list(Creact.pos_ID)
KreactList = list(Kreact.pos_ID)
print(len(CreactList))
print(len(KreactList))

C2018 = list(Cys2018)
K2018 = list(Lys2018) 

1541
4579


In [14]:
# CYS reactivity overlap with CYS2018
overlapCysreact = set(C2018).intersection(CreactList)
overlapLysreact = set(K2018).intersection(KreactList)
print(len(overlapCysreact))
print(len(overlapLysreact))

1492
4504


In [15]:
# label the original data frame with new column whether its in CCDS 2018 or not
def addcolumnconditional(mapList, df, dfcol, newcol): 
    mendel = []
    for g in df[dfcol]:
        if g in mapList: 
            mendel.append("True")
        else:
            mendel.append("False")
    df.loc[:,newcol] = mendel
    return df
CreactOverlapped = addcolumnconditional(overlapCysreact, Creact, "pos_ID", "2018pos_ID")
KreactOverlapped = addcolumnconditional(overlapLysreact, Kreact, "pos_ID", "2018pos_ID")


In [16]:
# seperating True and False then saving False
CreactTRUEonly = CreactOverlapped[CreactOverlapped["2018pos_ID"] == "True"]
CreactFALSEonly = CreactOverlapped[CreactOverlapped["2018pos_ID"] == "False"]
print(CreactTRUEonly.shape)
print(CreactFALSEonly.shape)

(1492, 3)
(49, 3)


In [17]:
# seperating True and False then saving False
KreactTRUEonly = KreactOverlapped[KreactOverlapped["2018pos_ID"] == "True"]
KreactFALSEonly = KreactOverlapped[KreactOverlapped["2018pos_ID"] == "False"]
print(KreactTRUEonly.shape)
print(KreactFALSEonly.shape)

(4504, 3)
(75, 3)


In [46]:
# saving rows not found in 2018 file
CreactFALSEonly.to_csv("Cys_reactivity_notFoundin2018positions_49pos.csv",index=False)
KreactFALSEonly.to_csv("Lys_reactivity_notFoundin2018positions_75pos.csv",index=False)

# redoing all above analysis with pan/target position files


In [18]:
# finding overlap between cys and lysine reactivity with 2018 positions

# making list of react pos_ID column
CpanList = list(Cpan.pos_ID)
KpanList = list(Kpan.pos_ID)
print(len(CpanList))
print(len(KpanList))


6157
8360


In [19]:
# CYS reactivity overlap with CYS2018
overlapCyspan = set(C2018).intersection(CpanList)
overlapLyspan = set(K2018).intersection(KpanList)
print(len(overlapCyspan))
print(len(overlapLyspan))

5991
8220


In [20]:
# label the original data frame with new column whether its in CCDS 2018 or not
def addcolumnconditional(mapList, df, dfcol, newcol): 
    mendel = []
    for g in df[dfcol]:
        if g in mapList: 
            mendel.append("True")
        else:
            mendel.append("False")
    df.loc[:,newcol] = mendel
    return df
CpanOverlapped = addcolumnconditional(overlapCyspan, Cpan, "pos_ID", "2018pos_ID")
KpanOverlapped = addcolumnconditional(overlapLyspan, Kpan, "pos_ID", "2018pos_ID")

In [21]:
# seperating True and False then saving False
CpanTRUEonly = CpanOverlapped[CpanOverlapped["2018pos_ID"] == "True"]
CpanFALSEonly = CpanOverlapped[CpanOverlapped["2018pos_ID"] == "False"]
print(CpanTRUEonly.shape)
print(CpanFALSEonly.shape)

(5991, 3)
(166, 3)


In [22]:
# seperating True and False then saving False
KpanTRUEonly = KpanOverlapped[KpanOverlapped["2018pos_ID"] == "True"]
KpanFALSEonly = KpanOverlapped[KpanOverlapped["2018pos_ID"] == "False"]
print(KpanTRUEonly.shape)
print(KpanFALSEonly.shape)

(8220, 3)
(140, 3)


In [61]:
# saving rows not found in 2018 file
CpanFALSEonly.to_csv("Cys_panTarget_notFoundin2018positions_166pos.csv",index=False)
KpanFALSEonly.to_csv("Lys_panTarget_notFoundin2018positions_140pos.csv",index=False)

In [23]:
CpanTRUEonly.reset_index(drop=True,inplace=True)
CreactTRUEonly.reset_index(drop=True,inplace=True)

KpanTRUEonly.reset_index(drop=True,inplace=True)
KreactTRUEonly.reset_index(drop=True,inplace=True)

# [TODO 2] format Cys reactivity data by taking the average of values in column, then plot the reactivity of all Cys and Lys values to define what thresholds will be LOW-MEDIUM-HIGH cutoff

In [24]:
# turning series into list
listCreact = list(CreactTRUEonly.reactivity)

# turning list into list of lists
llcreact = [x.split() for x in listCreact]

In [77]:
llcreact

[['20.00', '1/2/6'],
 ['5.74', '1/1/6'],
 ['20.00', '1/2/6'],
 ['0.90', '1/1/6'],
 ['1.86', '5.41', '2/4/6'],
 ['2.26', '1.88', '2.08', '2.29', '2.01', '2.12', '6/6/6'],
 ['5.79', '5.43', '2/2/6'],
 ['8.09', '1/1/6'],
 ['6.83', '3.16', '4.37', '3.99', '5.68', '5.81', '6/6/6'],
 ['7.56', '6.38', '2/2/6'],
 ['3.16', '3.11', '2.62', '3/3/6'],
 ['3.16', '3.11', '2.62', '3/3/6'],
 ['5.2', '5.73', '2/2/6'],
 ['1.38', '1/1/6'],
 ['3.96', '10.92', '2/3/6'],
 ['4.56', '7.27', '2/2/6'],
 ['8.51', '8.07', '2/2/6'],
 ['8.66', '1/1/6'],
 ['3.97', '1/1/6'],
 ['8.52', '1/2/6'],
 ['0.88', '1/1/6'],
 ['9.14', '3.58', '6.06', '5.74', '4/4/6'],
 ['3.63', '5.46', '5.29', '5.3', '4/4/6'],
 ['5.32', '1/1/6'],
 ['7.78', '1/1/6'],
 ['4.28', '1/1/6'],
 ['6.87', '6.63', '5.73', '5.77', '4/5/6'],
 ['3.99', '3.0', '4.19', '3/3/6'],
 ['4.75', '1/1/6'],
 ['5.17', '5.2', '2/2/6'],
 ['4.82', '1/1/6'],
 ['1.25', '1.22', '1.12', '1.17', '1.1', '1.02', '6/6/6'],
 ['20.00', '1/1/6'],
 ['5.40', '1/1/6'],
 ['5.03', '5.98',

In [25]:
for sublist in llcreact:
    del sublist[-1]
llcreact
# removed last index of sublist which contained the / value

[['20.00'],
 ['5.74'],
 ['20.00'],
 ['0.90'],
 ['1.86', '5.41'],
 ['2.26', '1.88', '2.08', '2.29', '2.01', '2.12'],
 ['5.79', '5.43'],
 ['8.09'],
 ['6.83', '3.16', '4.37', '3.99', '5.68', '5.81'],
 ['7.56', '6.38'],
 ['3.16', '3.11', '2.62'],
 ['3.16', '3.11', '2.62'],
 ['5.2', '5.73'],
 ['1.38'],
 ['3.96', '10.92'],
 ['4.56', '7.27'],
 ['8.51', '8.07'],
 ['8.66'],
 ['3.97'],
 ['8.52'],
 ['0.88'],
 ['9.14', '3.58', '6.06', '5.74'],
 ['3.63', '5.46', '5.29', '5.3'],
 ['5.32'],
 ['7.78'],
 ['4.28'],
 ['6.87', '6.63', '5.73', '5.77'],
 ['3.99', '3.0', '4.19'],
 ['4.75'],
 ['5.17', '5.2'],
 ['4.82'],
 ['1.25', '1.22', '1.12', '1.17', '1.1', '1.02'],
 ['20.00'],
 ['5.40'],
 ['5.03', '5.98'],
 ['5.03', '5.98'],
 ['7.87'],
 ['3.8', '2.29'],
 ['1.23'],
 ['6.86', '5.91'],
 ['9.14'],
 ['7.63'],
 ['3.54'],
 ['9.56', '3.75', '6.62', '4.81', '5.86', '13.0'],
 ['9.09', '3.66', '7.33', '7.74', '6.85', '6.47'],
 ['6.03', '3.37', '4.45', '4.3', '4.41', '4.47'],
 ['6.12', '4.53'],
 ['3.66', '7.0'],
 ['4

In [26]:
# getting average of lists in list
avgg = []
for sublist in llcreact:
    test_list = [float(i) for i in sublist]
    s = sum(test_list)
    l = len(sublist)
    a = s/l
    avgg.append(a)
avgg

[20.0,
 5.74,
 20.0,
 0.9,
 3.6350000000000002,
 2.106666666666667,
 5.609999999999999,
 8.09,
 4.973333333333334,
 6.97,
 2.9633333333333334,
 2.9633333333333334,
 5.465,
 1.38,
 7.4399999999999995,
 5.914999999999999,
 8.29,
 8.66,
 3.97,
 8.52,
 0.88,
 6.130000000000001,
 4.92,
 5.32,
 7.78,
 4.28,
 6.25,
 3.7266666666666666,
 4.75,
 5.1850000000000005,
 4.82,
 1.1466666666666665,
 20.0,
 5.4,
 5.505000000000001,
 5.505000000000001,
 7.87,
 3.045,
 1.23,
 6.385,
 9.14,
 7.63,
 3.54,
 7.266666666666666,
 6.8566666666666665,
 4.505,
 5.325,
 5.33,
 4.24,
 6.88,
 4.46,
 6.0,
 6.175,
 6.175,
 7.845,
 9.01,
 5.8933333333333335,
 4.49,
 5.865,
 8.15,
 6.586,
 2.12,
 6.63,
 6.835,
 1.67,
 20.0,
 5.545,
 9.3,
 6.64,
 1.61,
 4.55,
 7.55,
 5.95,
 6.755000000000001,
 1.87,
 7.8,
 3.16,
 8.09,
 5.263333333333333,
 0.8875,
 1.92,
 5.47,
 5.640000000000001,
 5.94,
 5.9775,
 4.86,
 9.8,
 9.11,
 3.2,
 3.2,
 7.07,
 4.375,
 7.48,
 3.47,
 5.6,
 5.775,
 4.275,
 1.01,
 4.933999999999999,
 2.9,
 2.9,
 6.

In [27]:
# formating decimal place
avgf = ["{0:.2f}".format(x) for x in avgg]
avgf

['20.00',
 '5.74',
 '20.00',
 '0.90',
 '3.64',
 '2.11',
 '5.61',
 '8.09',
 '4.97',
 '6.97',
 '2.96',
 '2.96',
 '5.46',
 '1.38',
 '7.44',
 '5.91',
 '8.29',
 '8.66',
 '3.97',
 '8.52',
 '0.88',
 '6.13',
 '4.92',
 '5.32',
 '7.78',
 '4.28',
 '6.25',
 '3.73',
 '4.75',
 '5.19',
 '4.82',
 '1.15',
 '20.00',
 '5.40',
 '5.51',
 '5.51',
 '7.87',
 '3.04',
 '1.23',
 '6.38',
 '9.14',
 '7.63',
 '3.54',
 '7.27',
 '6.86',
 '4.50',
 '5.33',
 '5.33',
 '4.24',
 '6.88',
 '4.46',
 '6.00',
 '6.17',
 '6.17',
 '7.84',
 '9.01',
 '5.89',
 '4.49',
 '5.87',
 '8.15',
 '6.59',
 '2.12',
 '6.63',
 '6.83',
 '1.67',
 '20.00',
 '5.54',
 '9.30',
 '6.64',
 '1.61',
 '4.55',
 '7.55',
 '5.95',
 '6.76',
 '1.87',
 '7.80',
 '3.16',
 '8.09',
 '5.26',
 '0.89',
 '1.92',
 '5.47',
 '5.64',
 '5.94',
 '5.98',
 '4.86',
 '9.80',
 '9.11',
 '3.20',
 '3.20',
 '7.07',
 '4.38',
 '7.48',
 '3.47',
 '5.60',
 '5.78',
 '4.28',
 '1.01',
 '4.93',
 '2.90',
 '2.90',
 '6.54',
 '8.10',
 '5.04',
 '2.96',
 '4.29',
 '1.06',
 '6.92',
 '5.50',
 '5.68',
 '10.0

In [28]:
print(len(avgf))

1492


In [29]:
# convert this into a column for CreactTRUE df
avgR = pd.DataFrame(np.array(avgf).reshape(-1,1))
avgR.columns = ['avgReactivity']
avgR

avgReactivity
0            20.00
1             5.74
2            20.00
3             0.90
4             3.64
5             2.11
6             5.61
7             8.09
8             4.97
9             6.97
10            2.96
11            2.96
12            5.46
13            1.38
14            7.44
15            5.91
16            8.29
17            8.66
18            3.97
19            8.52
20            0.88
21            6.13
22            4.92
23            5.32
24            7.78
25            4.28
26            6.25
27            3.73
28            4.75
29            5.19
...            ...
1462          5.37
1463          5.70
1464         20.00
1465          4.32
1466          6.51
1467          3.87
1468          5.80
1469          5.96
1470          2.11
1471          5.45
1472          4.65
1473          5.17
1474         10.32
1475          3.73
1476          1.09
1477          2.84
1478          2.84
1479          3.06
1480          3.06
1481          4.44
1482          5.04
1483          4.68
1484          3.62
1485          3.62
1486          1.28
1487          6.06
1488          4.40
1489          8.48
1490          4.26
1491          2.08

[1492 rows x 1 columns]

In [30]:
C2reactTRUEonly = pd.concat([CreactTRUEonly, avgR],axis=1)


In [31]:
Cfinalreact = C2reactTRUEonly[['pos_ID', 'avgReactivity']].copy()
Cfinalreact.head(3)

pos_ID avgReactivity
0   Q9Y6W6_C177         20.00
1    Q9Y6M9_C42          5.74
2  Q9Y6J0_C1197         20.00

In [32]:
Cfinalreact.columns = ['pos_ID','reactivity']

In [33]:
Kfinalreact = KreactTRUEonly[['pos_ID','reactivity']].copy()
Kfinalreact.head(3)

pos_ID  reactivity
0  P52758_K117         0.1
1   P02766_K35         0.1
2   P51572_K79         0.2

In [34]:
Kfinalreact.reactivity = pd.to_numeric(Kfinalreact.reactivity)
Cfinalreact.reactivity = pd.to_numeric(Cfinalreact.reactivity)

In [35]:
Cfinalreact = Cfinalreact.sort_values(by=['reactivity'])
Kfinalreact = Kfinalreact.sort_values(by=['reactivity'])

In [36]:
allreact = Kfinalreact.append(Cfinalreact,ignore_index=True)

In [37]:
allreact.reactivity = pd.to_numeric(allreact.reactivity)
allreact.dtypes

pos_ID         object
reactivity    float64
dtype: object

In [38]:
allreact = allreact.sort_values(by=['reactivity'])
allreact.reset_index(drop=True,inplace=True)

In [39]:
allreact.drop(['Qgroup'],inplace=True, axis=1)

KeyError: "['Qgroup'] not found in axis"

# [TODO 3] plot distribution of reactivity in LYS and CYS datasets

An accurate graphical representation of the distribution of numeric data = histogram

In [40]:
print(allreact.describe())

        reactivity
count  5996.000000
mean      7.427051
std       3.163972
min       0.100000
25%       5.000000
50%       8.700000
75%      10.000000
max      20.000000


In [144]:
# adding new threshold by old numbers from nature paper      
allreact.loc[allreact.reactivity <= 2, 'Threshold'] = 'High' 
allreact.loc[(allreact.reactivity > 2) & (allreact.reactivity <= 5), 'Threshold'] = 'Medium' 
allreact.loc[allreact.reactivity > 5, 'Threshold'] = 'Low'

from collections import Counter
thresholdf = Counter(list(allreact.Threshold))
thresdic = dict(thresholdf)
thresdic

In [145]:
from collections import Counter
thresholdf = Counter(list(allreact.Threshold))
thresdic = dict(thresholdf)
thresdic

{'High': 402, 'Medium': 1118, 'Low': 4476}

# HIGH MED LOW COUNTS of all reactivity data

## {'High': 402, 'Medium': 1118, 'Low': 4476}

### HIGH : <= 2
### MEDIUM : 2+ & <= 5
### LOW : 5+

In [146]:
allreact[allreact["Threshold"] == "Low"].describe()

reactivity
count  4476.000000
mean      8.924229
std       1.995162
min       5.010000
25%       7.387500
50%      10.000000
75%      10.000000
max      20.000000

In [148]:
allreact[allreact["Threshold"] == "Medium"].describe()

reactivity
count  1118.000000
mean      3.649177
std       0.874127
min       2.010000
25%       2.900000
50%       3.700000
75%       4.460000
max       5.000000

In [147]:
allreact[allreact["Threshold"] == "High"].describe()

reactivity
count  402.000000
mean     1.263607
std      0.464303
min      0.100000
25%      0.900000
50%      1.300000
75%      1.692500
max      2.000000

# THRESHOLDs for CYS and LYS data seperately

In [41]:
Cfinalreact.reset_index(drop=True,inplace=True)
Kfinalreact.reset_index(drop=True,inplace=True)

In [42]:
Kfinalreact.head(10)

pos_ID  reactivity
0   P52758_K117         0.1
1    P02766_K35         0.1
2    P51572_K79         0.2
3   P60900_K116         0.2
4    Q86WQ0_K28         0.2
5    Q9NZ45_K68         0.2
6   P08237_K678         0.3
7    O14737_K66         0.3
8    P23284_K41         0.3
9  P53396_K1018         0.3

In [43]:
print(Kfinalreact.describe())

        reactivity
count  4504.000000
mean      7.926732
std       2.999950
min       0.100000
25%       5.800000
50%      10.000000
75%      10.000000
max      10.000000


In [44]:
Cfinalreact.head(10)

pos_ID  reactivity
0  O14733_C202        0.74
1   P07237_C56        0.80
2   P78417_C32        0.84
3  Q6DCA0_C153        0.88
4  Q9Y4X0_C175        0.88
5   Q00796_C45        0.89
6   Q9UHQ1_C99        0.89
7  Q9Y6I4_C302        0.90
8  P38606_C138        0.90
9   Q8IY18_C91        0.92

In [45]:
print(Cfinalreact.describe())

        reactivity
count  1492.000000
mean      5.918633
std       3.168556
min       0.740000
25%       4.290000
50%       5.560000
75%       6.830000
max      20.000000


In [46]:
# adding new threshold by old numbers from nature paper      
Cfinalreact.loc[Cfinalreact.reactivity <= 2, 'Threshold'] = 'High' 
Cfinalreact.loc[(Cfinalreact.reactivity > 2) & (Cfinalreact.reactivity <= 5), 'Threshold'] = 'Medium' 
Cfinalreact.loc[Cfinalreact.reactivity > 5, 'Threshold'] = 'Low'


In [47]:

from collections import Counter
Cthres = Counter(list(Cfinalreact.Threshold))
Cdic = dict(Cthres)
Cdic

{'High': 97, 'Medium': 451, 'Low': 944}

In [48]:
clow=Cfinalreact[Cfinalreact["Threshold"] == "Low"].describe()

clow.columns=['C_LOW']
clow

C_LOW
count  944.000000
mean     7.382680
std      3.027032
min      5.010000
25%      5.710000
50%      6.440000
75%      7.840000
max     20.000000

In [49]:
cmed = Cfinalreact[Cfinalreact["Threshold"] == "Medium"].describe()
cmed.columns=['C_MED']
cmed

C_MED
count  451.000000
mean     3.839645
std      0.827942
min      2.010000
25%      3.185000
50%      4.000000
75%      4.550000
max      5.000000

In [50]:
chigh=Cfinalreact[Cfinalreact["Threshold"] == "High"].describe()
chigh.columns=['C_HIGH']
chigh

C_HIGH
count  97.000000
mean    1.336804
std     0.340152
min     0.740000
25%     1.060000
50%     1.320000
75%     1.640000
max     1.960000

In [51]:
CLM = clow.join(cmed)
CLMH = CLM.join(chigh)
print(CLMH)


            C_LOW       C_MED     C_HIGH
count  944.000000  451.000000  97.000000
mean     7.382680    3.839645   1.336804
std      3.027032    0.827942   0.340152
min      5.010000    2.010000   0.740000
25%      5.710000    3.185000   1.060000
50%      6.440000    4.000000   1.320000
75%      7.840000    4.550000   1.640000
max     20.000000    5.000000   1.960000


In [53]:
# adding new threshold by old numbers from nature paper      
Kfinalreact.loc[Kfinalreact.reactivity <= 2, 'Threshold'] = 'High' 
Kfinalreact.loc[(Kfinalreact.reactivity > 2) & (Kfinalreact.reactivity <= 5), 'Threshold'] = 'Medium' 
Kfinalreact.loc[Kfinalreact.reactivity > 5, 'Threshold'] = 'Low'

In [54]:

from collections import Counter
kthre = Counter(list(Kfinalreact.Threshold))
kdic = dict(kthre)
kdic

{'High': 305, 'Medium': 667, 'Low': 3532}

In [55]:
klow = Kfinalreact[Kfinalreact["Threshold"] == "Low"].describe()
klow.columns = ['K_LOW']
klow

K_LOW
count  3532.000000
mean      9.336240
std       1.338915
min       5.100000
25%       9.900000
50%      10.000000
75%      10.000000
max      10.000000

In [57]:
kmed = Kfinalreact[Kfinalreact["Threshold"] == "Medium"].describe()
kmed.columns = ['K_MED']
kmed

K_MED
count  667.000000
mean     3.520390
std      0.881579
min      2.100000
25%      2.700000
50%      3.500000
75%      4.300000
max      5.000000

In [58]:
khigh=Kfinalreact[Kfinalreact["Threshold"] == "High"].describe()
khigh.columns=['K_HIGH']
khigh

K_HIGH
count  305.000000
mean     1.240328
std      0.495552
min      0.100000
25%      0.900000
50%      1.300000
75%      1.700000
max      2.000000

In [59]:
klm = klow.join(kmed)
klmh = klm.join(khigh)
print(klmh)

             K_LOW       K_MED      K_HIGH
count  3532.000000  667.000000  305.000000
mean      9.336240    3.520390    1.240328
std       1.338915    0.881579    0.495552
min       5.100000    2.100000    0.100000
25%       9.900000    2.700000    0.900000
50%      10.000000    3.500000    1.300000
75%      10.000000    4.300000    1.700000
max      10.000000    5.000000    2.000000


In [60]:
CpanTRUEonly.reset_index(drop=True,inplace=True)
CreactTRUEonly.reset_index(drop=True,inplace=True)

KpanTRUEonly.reset_index(drop=True,inplace=True)
KreactTRUEonly.reset_index(drop=True,inplace=True)

In [61]:
def checkColumnValues(df, col):
    print(df[col].value_counts().reset_index().rename(columns={'index':col, col:'Count'}))

In [62]:
checkColumnValues(CpanTRUEonly, "labelType")

       labelType  Count
0  Iodoacetamide   5254
1         Target    737


In [63]:
print(CpanTRUEonly.shape)
print(len(set(CpanTRUEonly.pos_ID)))

(5991, 3)
5991


In [64]:
checkColumnValues(KpanTRUEonly,"labelType")

  labelType  Count
0     IodoL   8102
1   TargetL    118


In [65]:
print(KpanTRUEonly.shape)
print(len(set(KpanTRUEonly.pos_ID)))

(8220, 3)
8220


In [66]:
Cpanlist = list(CpanTRUEonly.pos_ID)
Creaclist = list(CreactTRUEonly.pos_ID)

Kpanlist = list(KpanTRUEonly.pos_ID)
Kreaclist = list(KreactTRUEonly.pos_ID)

C = set(Cpanlist + Creaclist)
K = set(Kpanlist + Kreaclist)

print(len(C))
print(len(K))

6315
9162


In [68]:
# saving ever laveled columns
CC = list(C)
KK = list(K)

Cever = pd.DataFrame(np.array(CC).reshape(-1,1))
Cever.columns = ["Ever_labeled_Cys_set"]

Kever = pd.DataFrame(np.array(KK).reshape(-1,1))
Kever.columns = ["Ever_labeled_Lys_set"]



In [69]:
CpanTRUEonly.labelType.replace("Iodoacetamide","panReactive",inplace=True)


/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py:5890: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


In [70]:
checkColumnValues(CpanTRUEonly,"labelType")

     labelType  Count
0  panReactive   5254
1       Target    737


In [71]:
KpanTRUEonly.labelType.replace("IodoL","panReactive",inplace=True)
KpanTRUEonly.labelType.replace("TargetL","Target",inplace=True)

In [72]:
checkColumnValues(KpanTRUEonly,"labelType")

     labelType  Count
0  panReactive   8102
1       Target    118


In [73]:
Cfinalreact= Cfinalreact.assign(subset="Cysteine")
Cever = Cever.assign(subset="Cysteine")
CpanTRUEonly = CpanTRUEonly.assign(subset="Cysteine")

Kfinalreact = Kfinalreact.assign(subset="Lysine")
Kever = Kever.assign(subset="Lysine")
KpanTRUEonly = KpanTRUEonly.assign(subset="Lysine")

```python
csplit = Cever["Ever_labeled_Cys_set"].str.split("_",n=1,expand=True)
lsplit = Kever["Ever_labeled_Lys_set"].str.split("_",n=1,expand=True)
csplit.columns = ["entry", "position"]
lsplit.columns = ["entry","position"]
```

In [104]:
def splitUKBentry(df, dfcol):
    split = df[dfcol].str.split("_", n=1, expand=True)
    split.columns= ["entry","pos"]
    split2 = pd.concat([df, split], axis =1)
    return split2


In [105]:
Cever2 = splitUKBentry(Cever, "Ever_labeled_Cys_set")
Kever2 = splitUKBentry(Kever, "Ever_labeled_Lys_set")
Cfinalreact2 = splitUKBentry(Cfinalreact,"pos_ID")
Kfinalreact2 = splitUKBentry(Kfinalreact,"pos_ID")

CpanTRUEonly2 = splitUKBentry(CpanTRUEonly,"pos_ID")
KpanTRUEonly2 = splitUKBentry(KpanTRUEonly,"pos_ID")

In [106]:
Cever2.head(3)

Ever_labeled_Cys_set    subset   entry   pos
0          Q68E01_C596  Cysteine  Q68E01  C596
1           P30101_C57  Cysteine  P30101   C57
2          Q9UBQ7_C216  Cysteine  Q9UBQ7  C216

In [107]:
Kever2.head(3)

Ever_labeled_Lys_set  subset   entry   pos
0          Q09666_K632  Lysine  Q09666  K632
1          Q16740_K211  Lysine  Q16740  K211
2          P53367_K137  Lysine  P53367  K137

In [108]:
Cfinalreact2.head(3)

pos_ID  reactivity Threshold    subset   entry   pos
0  O14733_C202        0.74      High  Cysteine  O14733  C202
1   P07237_C56        0.80      High  Cysteine  P07237   C56
2   P78417_C32        0.84      High  Cysteine  P78417   C32

In [109]:
Kfinalreact2.head(3)

pos_ID  reactivity Threshold  subset   entry   pos
0  P52758_K117         0.1      High  Lysine  P52758  K117
1   P02766_K35         0.1      High  Lysine  P02766   K35
2   P51572_K79         0.2      High  Lysine  P51572   K79

In [110]:
CpanTRUEonly2.head(3)

pos_ID    labelType    subset   entry   pos
0  P26641_C266  panReactive  Cysteine  P26641  C266
1  P07900_C374  panReactive  Cysteine  P07900  C374
2  P23528_C139  panReactive  Cysteine  P23528  C139

In [111]:
KpanTRUEonly2.head(3)

pos_ID    labelType  subset   entry    pos
0  A0AVT1_K1014  panReactive  Lysine  A0AVT1  K1014
1   A0AVT1_K544  panReactive  Lysine  A0AVT1   K544
2    A0AVT1_K86  panReactive  Lysine  A0AVT1    K86

# SAVING ALL FILES!

In [112]:
Cever2.to_csv("Cys_ever_labeled_set_6315.csv",index=False)
Kever2.to_csv("Lys_ever_labeled_set_9162.csv",index=False)

Cfinalreact2.to_csv("Cys_reactive_LMHset_1492.csv",index=False)
Kfinalreact2.to_csv("Lys_reactive_LMHset_4504.csv",index=False)

CpanTRUEonly2.to_csv("Cys_panTar_5991.csv",index=False)
KpanTRUEonly2.to_csv("Lys_panTar_8220.csv",index=False)

In [113]:
KpanTRUEonly.shape

(8220, 3)

In [75]:
Cfinalreact.head(3)

pos_ID  reactivity Threshold    subset
0  O14733_C202        0.74      High  Cysteine
1   P07237_C56        0.80      High  Cysteine
2   P78417_C32        0.84      High  Cysteine

In [76]:
Cever.head(3)

Ever_labeled_Cys_set    subset
0          Q68E01_C596  Cysteine
1           P30101_C57  Cysteine
2          Q9UBQ7_C216  Cysteine

In [77]:
CpanTRUEonly.drop(['2018pos_ID'],inplace=True, axis=1)

In [78]:
KpanTRUEonly.drop(['2018pos_ID'],inplace=True, axis=1)

In [79]:
CpanTRUEonly.head(3)

pos_ID    labelType    subset
0  P26641_C266  panReactive  Cysteine
1  P07900_C374  panReactive  Cysteine
2  P23528_C139  panReactive  Cysteine

In [103]:
KpanTRUEonly.head(3)

pos_ID    labelType  subset
0  A0AVT1_K1014  panReactive  Lysine
1   A0AVT1_K544  panReactive  Lysine
2    A0AVT1_K86  panReactive  Lysine

In [81]:
Kever.head(3)

Ever_labeled_Lys_set  subset
0          Q09666_K632  Lysine
1          Q16740_K211  Lysine
2          P53367_K137  Lysine

In [82]:
Kfinalreact.head(3)

pos_ID  reactivity Threshold  subset
0  P52758_K117         0.1      High  Lysine
1   P02766_K35         0.1      High  Lysine
2   P51572_K79         0.2      High  Lysine

---
---
---
---

# looking at nonRedundant ever labeled set, which genes have been labeled the most:


In [83]:
csplit = Cever["Ever_labeled_Cys_set"].str.split("_",n=1,expand=True)
lsplit = Kever["Ever_labeled_Lys_set"].str.split("_",n=1,expand=True)
csplit.columns = ["entry", "position"]
lsplit.columns = ["entry","position"]

In [84]:
allGenes = list(csplit.entry) + list(lsplit.entry)
allG = set(allGenes)
print(len(allG))

3991


In [86]:
csplit.head(3)

entry position
0  Q68E01     C596
1  P30101      C57
2  Q9UBQ7     C216

In [87]:
lsplit.head(3)

entry position
0  Q09666     K632
1  Q16740     K211
2  P53367     K137

In [92]:
# enteries overlapping between Cys and Lys
c_entry = list(csplit['entry'])
k_entry = list(lsplit['entry'])
c_set = set(c_entry)
k_set = set(k_entry)
print(len(c_set))
print(len(k_set))

2939
2635


In [96]:
len(list(c_set & k_set))

1583

In [98]:
inter = c_set.intersection(k_set)
len(inter) # overlapping entries between cys and lys data

1583

In [99]:
KminusC = k_set - c_set
CminusK = c_set - k_set
print(len(KminusC))
print(len(CminusK))

1052
1356


In [100]:
1583 + 1052 # matched unique lys entry

2635

In [101]:
1583 + 1356 # matched unique cys entry

2939

In [ ]:
# alleverlabeled = csplit.append(lsplit, ignore_index = True)
alleverlabeled

In [89]:
# count how many times each gene has EVER been labeled by C or K (all C and K positions are unique!)
togethergenecount = alleverlabeled.loc[:,'entry'].value_counts()
togethergenecount

Q09666    112
P21333     75
O75369     65
Q15149     58
P35579     56
P13639     43
Q9Y490     42
P07900     36
P49327     36
P00558     35
P31948     32
P14618     32
P07237     32
P10809     31
P08238     31
O43707     30
P02545     30
P68104     29
Q14204     29
Q01813     29
P11586     29
P06733     27
P11142     27
O75083     26
Q99832     25
Q00610     25
O00410     25
P30101     25
Q96HE7     24
Q7Z6Z7     24
         ... 
Q9H2U1      1
Q5T2E6      1
Q5JPI9      1
Q8TDG2      1
Q9Y697      1
P28331      1
Q9UJ68      1
Q15836      1
Q96LB3      1
P62736      1
O14880      1
Q5SRE7      1
Q96AD5      1
Q9UKT9      1
Q9BQ90      1
Q96JM3      1
P69905      1
P14923      1
Q96KA5      1
Q0VFZ6      1
Q8N4V2      1
Q9UJU2      1
Q8NI08      1
Q13535      1
Q96PU5      1
Q7L5N7      1
Q9H8M7      1
Q8NBX0      1
Q07157      1
Q8N0T1      1
Name: entry, Length: 3991, dtype: int64

In [90]:
togethergenecount.to_csv("CYS_LYS_EverLabeled_gene_valueCounts_3991.csv")

In [221]:
# count how many times each gene has a C pos or K pos in file
cgenecount = csplit.loc[:,'entry'].value_counts()
lgenecount = lsplit.loc[:,'entry'].value_counts()

In [224]:
cgenecount.to_csv("CysEverLabeled_gene_valueCounts_2939.csv")
lgenecount.to_csv("LysEverLabeled_gene_valueCounts_2635.csv")

# finding genes that have most labeleing, want to count # of LOW MED HIGH

In [227]:
csplit = Cfinalreact["pos_ID"].str.split("_",n=1,expand=True)
lsplit = Kfinalreact["pos_ID"].str.split("_",n=1,expand=True)
csplit.columns = ["entry", "position"]
lsplit.columns = ["entry","position"]
cthr = Cfinalreact['Threshold']
kthr = Kfinalreact['Threshold']

Cpy = pd.concat([csplit,cthr],axis=1)
Kpy = pd.concat([lsplit,kthr],axis=1)



In [234]:
Cgroups = Cpy.groupby(['entry','Threshold']).count()
Kgroups = Kpy.groupby(['entry','Threshold']).count()

In [236]:
Cgroups.to_csv("Cys_entry_Threshold_grouping_PymolStruc.csv")
Kgroups.to_csv("Lys_entry_Threshold_grouping_PymolStruc.csv")

In [237]:
Cgroups

position
entry  Threshold          
A0AVT1 Medium            2
A0MZ66 Low               1
A6NDG6 Medium            1
A6NDU8 Low               1
A6NED2 High              1
A6NHL2 Medium            3
A6NNF4 Low               1
A9UHW6 Medium            1
O00170 Medium            2
O00244 Medium            2
O00267 High              1
O00268 Medium            1
O00273 Low               2
O00299 Low               2
       Medium            2
O00303 Low               1
O00429 Low               1
       Medium            1
O00487 Medium            1
O00506 Low               2
O00515 Low               1
O00541 Medium            1
O00743 Medium            1
O00764 Low               1
O14545 Low               1
O14556 Low               1
O14579 Medium            1
O14733 High              1
       Medium            1
O14744 Low               1
...                    ...
Q9Y3A3 Low               1
Q9Y3C6 Low               1
Q9Y3D0 High              1
Q9Y3D2 Low               1
       Medium            1
Q9Y3E2 Medium            2
Q9Y3F4 Low               1
Q9Y478 Medium            1
Q9Y490 Low               1
Q9Y4P1 Low               1
       Medium            1
Q9Y4R8 Low               1
Q9Y4X0 High              1
Q9Y508 Low               1
Q9Y5B9 Medium            1
Q9Y5K6 Low               1
Q9Y5N6 Low               1
Q9Y5P6 Low               1
Q9Y5R8 Low               1
Q9Y5S2 High              1
Q9Y5S9 Low               1
Q9Y5Y2 Low               1
       Medium            3
Q9Y617 Low               2
Q9Y696 Medium            1
Q9Y6E0 Medium            1
Q9Y6I4 High              1
Q9Y6J0 Low               1
Q9Y6M9 Low               1
Q9Y6W6 Low               1

[1130 rows x 1 columns]

---
---
---
---